pages/api/events/class/index.ts

```diff
@@ -57,28 +87,81 @@ const classEventHandler: NextApiHandler = async (req: NextApiRequest, res: NextA
               minute: startTime.minute,
               second: startTime.second,
             })
-            .setZone(newEvent.timeZone);
+            .setZone(newEvent.timeZone, { keepLocalTime: true });
 
           const dateEnd = DateTime.fromJSDate(dateWithoutTime)
             .set({
               hour: endTime.hour,
               minute: endTime.minute,
               second: endTime.second,
             })
-            .setZone(newEvent.timeZone);
-
-          intervals.push(Interval.fromDateTimes(dateStart, dateEnd));
+            .setZone(newEvent.timeZone, { keepLocalTime: true });
+
+          const interval = Interval.fromDateTimes(dateStart, dateEnd);
+          intervals.push(interval);
+
+          const weekDay = dateStart.weekday;
+          if (!seenWeekDays.has(weekDay)) {
+            seenWeekDays.add(weekDay);
+            const weekDayStr = indexToWeekdays[weekDay] as Weekdays;
+            if (weekDayStr != "sun") {
+              intervalsByWeekDay[weekDayStr] = interval;
+            }
```

Comment:
what exactly is being done here? Can we add some comments just to be a bit clear